sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1133,17 +1135,19 @@ static int sof_card_dai_links_create(struct snd_soc_card *card)
 		hdmi_num = SOF_PRE_TGL_HDMI_COUNT;
 
 	/* enable dmic01 & dmic16k */
-	if (sof_sdw_quirk & SOC_SDW_PCH_DMIC || mach_params->dmic_num) {
-		if (ctx->ignore_internal_dmic)
-			dev_warn(dev, "Ignoring PCH DMIC\n");
-		else
-			dmic_num = 2;
+	if (ctx->ignore_internal_dmic) {
+		dev_warn(dev, "Ignoring internal DMIC\n");
+		mach_params->dmic_num = 0;
+	} else if (mach_params->dmic_num) {
+		dmic_num = 2;
```

Comment:
Regardless of the value of `mach_params->dmic_num` dmic_num is wired to **2**? This might be obvious, but it is really not to me, especially when looking at the next else branch where both dmic_num is set to two.

Reply:
> @bardliao is mach_params->dmic_num 0 by default? If not, what could it be set to before we get to line 1140?

[mach->mach_params.dmic_num = check_dmic_num(sdev);](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.c#L1308) is set when selecting machine driver which is before machine driver is probed.

Reply:
sorry to keep prolonging the discussion @bardliao . But if mach_params->dmic is 0 and sof_sdw_quirk & SOC_SDW_PCH_DMIC is set, why is the maach_params->dmic_num set to 2? and when will this case happen?

Reply:
> sorry to keep prolonging the discussion @bardliao . But if mach_params->dmic is 0 and sof_sdw_quirk & SOC_SDW_PCH_DMIC is set, why is the maach_params->dmic_num set to 2? and when will this case happen?

The case happens when it is a Chrome book. The coreboot doesn't provide the dmic information and we don't have a way to know the dmic channels. So set it to 2.